### PR TITLE
Adding gcp typescript example for deploy of image to gcr and cloud run

### PR DESCRIPTION
**Pull request by danfhernandez**
Would be great to get someone to check this out and let me know if there is a better way to do this.. 

Goal is to build and deploy a docker image for an app in one project, and then in a separate project, check for any changes to the image and redeploy.

### CODE REVIEW

**Comment by danfhernandez** [comment]
@mikhailshilkov @stack72 would be great to get a review of this :)

File: gcp-ts-docker-gcr-cloudrun/README.md [hunk 1]
```diff
@@ -0,0 +1,136 @@
+[![Deploy](https://get.pulumi.com/new/button.svg)](https://app.pulumi.com/new)
+
+# Docker Build and Push to GCR and Deploy to Google Cloud Run using separate projects
+
+An example of building a custom Docker image and pushing it into a Google Cloud Container Registry and then in a seperate project deploying that image with the Google Cloud Run service using TypeScript.
```
Comment (mikhailshilkov) [inline gcp-ts-docker-gcr-cloudrun/README.md:5]:
```suggestion
An example of building a custom Docker image and pushing it into a Google Cloud Container Registry and then in a separate project deploying that image with the Google Cloud Run service using TypeScript.
```

File: gcp-ts-docker-gcr-cloudrun/cloud-run-deploy/index.ts [hunk 2]
```diff
@@ -0,0 +1,59 @@
+import * as pulumi from "@pulumi/pulumi";
```
Comment (mikhailshilkov) [inline gcp-ts-docker-gcr-cloudrun/cloud-run-deploy/index.ts:1]: Add the copyright header

File: gcp-ts-docker-gcr-cloudrun/cloud-run-deploy/index.ts [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+import * as gcp from "@pulumi/gcp";
+import * as docker from "@pulumi/docker";
+
+const location = gcp.config.region || "us-central1";
+
+const config = new pulumi.Config()
```
Comment (mikhailshilkov) [inline gcp-ts-docker-gcr-cloudrun/cloud-run-deploy/index.ts:7]: You are missing a bunch of semicolons in code. We use linter for examples, so please run the linter until you have zero warnings there.

File: gcp-ts-docker-gcr-cloudrun/cloud-run-deploy/index.ts [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+// Used to get the image from the google cloud registry.  Output is required to make sure that the provider is in sync with this call. 
+const registryImage = pulumi.output(
+    docker.getRegistryImage({
+    name: `gcr.io/${gcp.config.project}/${imageName}:latest`,
+}, {provider: gcrDockerProvider}));
+
+
+// Using the value from the registryImage to pull the image if it's new, pullTriggers looks for a new sha. 
+var dockerImage = registryImage.apply(r => new docker.RemoteImage(`${imageName}-docker-image`, {
```
Comment (mikhailshilkov) [inline gcp-ts-docker-gcr-cloudrun/cloud-run-deploy/index.ts:29]: Do you need an `apply` here? Could you do `registryImage.name` the same way you do `registryImage.sha256Digest`?

File: gcp-ts-docker-gcr-cloudrun/cloud-run-deploy/index.ts [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+
+// Open the service to public unrestricted access
+const iamRuby = new gcp.cloudrun.IamMember("ruby-everyone", {
+    service: rubyService.name,
+    location,
+    role: "roles/run.invoker",
+    member: "allUsers",
+});
+
+export const rubyUrl = rubyService.status.url;
```
Comment (mikhailshilkov) [inline gcp-ts-docker-gcr-cloudrun/cloud-run-deploy/index.ts:59]: EOF

File: gcp-ts-docker-gcr-cloudrun/docker-build-push-gcr/index.ts [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+
+const myImage = new docker.Image(imageName, {
+    imageName: pulumi.interpolate`gcr.io/${gcp.config.project}/${imageName}:latest`,
+    build: {
+        context: "./app",
+    },
+});
+
+// Digest exported so it's easy to match updates happening in cloud run project
+export const digest = myImage.digest
```
Comment (mikhailshilkov) [inline gcp-ts-docker-gcr-cloudrun/docker-build-push-gcr/index.ts:19]: EOF

File: gcp-ts-docker-gcr-cloudrun/cloud-run-deploy/index.ts (hunk 4 above)
Comment (danfhernandez) [reply gcp-ts-docker-gcr-cloudrun/cloud-run-deploy/index.ts:29]: Yeah that's why I added the apply, weirdly got an error when trying that...

File: gcp-ts-docker-gcr-cloudrun/cloud-run-deploy/index.ts (hunk 3 above)
Comment (danfhernandez) [reply gcp-ts-docker-gcr-cloudrun/cloud-run-deploy/index.ts]: Should be good now?

**Pull request merged**